railgun/test/integration/controller_test.go

```diff
@@ -0,0 +1,52 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"bytes"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestHealthEndpoint intentionally includes a magic value. localhost:10254 is the default
+// controller health listen. This test should fail, and must be updated, if that default is changed.
```

Comment:
This is test validates two things:
- that the healthz endpoint returns a valid response
- that when somebody change the default value of the port, they also change the magic number in the test

The first part LGTM, the second one makes this test a change detector test. Implementing change detector tests is discouraged by many sources like [this](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html) because it adds repetition and a chore to update all affected sites by means of a find-replace operation.

Imagine that a new person comes to the team in a year, and wants to change the port number to some other value.
They will update the flag default and see the tests fail.
What will be the next thing they do? A find-replace of `10254` throughout the codebase.

At the same time, I acknowledge the need to express that this default is here for a reason, and changing it should not be taken lightly.

The best way to approach this without repetition I can imagine is by defining a `const` in the code with a comment explaining why changing this may be a bad idea, like:

```go
// HealthzPort configures the HTTP port on which manager will expose the health endpoint.
// Changing this will likely cause crash loops for all existing configurations, unless they also
// update the liveness probe.
const HealthzPort = 10254
```

and then using it in the test, verifying that the manager indeed has a probe endpoint as expected. It will work better than a test verifying that the same number is set in two places, because the person changing the number is (reasonably) guaranteed to acknowledge that warning.

We could optionally to a step further and get rid of the `/healthz` magic constant in the test, by defining manager's `ReadinessEndpointName` alongside `HealthProbeBindAddress` and therefore overriding the default. But I don't recommend doing that because it seems like overkill to me.

AI: replace the change detector with a reusable and well documented `const`

Reply:
0bb98af